nerdyrodent/VQGAN-CLIP
# #1 Add image metadata

Resolves #1

- [x] Add the ACMI source URL, author, description, and software to the generated image IPTC metadata field

## Testing

1. Run the notebook in Colab: https://colab.research.google.com/drive/1qBbpgsZR1UFhFhj-aDEF75tCCihkjOaI
1. Click the folder icon on the left and download one of the generated images
1. Use `exiftool` or similar to see the metadata

```
$ exiftool ~/Desktop/1.png 
ExifTool Version Number         : 12.30
File Name                       : 1.png
Directory                       : /Users/sloffler/Desktop
File Size                       : 175 KiB
File Modification Date/Time     : 2021:10:22 12:12:56+10:30
File Access Date/Time           : 2021:10:22 12:13:21+10:30
File Inode Change Date/Time     : 2021:10:22 12:13:19+10:30
File Permissions                : -rw-r--r--
File Type                       : PNG
File Type Extension             : png
MIME Type                       : image/png
Image Width                     : 592
Image Height                    : 592
Bit Depth                       : 8
Color Type                      : RGB
Compression                     : Deflate/Inflate
Filter                          : Adaptive
Interlace                       : Noninterlaced
Author                          : ACMI - Your museum of screen culture
Description                     : Image generated using VQGAN+CLIP from metadata: Tom Hughes|An old pirate sailing ship violently attacks|dolphins and seagulls at sea but encounters|trouble when bombed by another pirate ship.An|animated short entered to “Create a Stamp|Story” section of ACMI’s ‘Screen It’ media|making competition for Primary Schools.
Software                        : https://github.com/ACMILabs/ACMI-API-x-VQGAN-CLIP
Source                          : https://www.acmi.net.au/works/109561--tom-hughes/
Image Size                      : 592x592
Megapixels                      : 0.350
```
